device/uapi.go

```diff
@@ -99,33 +99,31 @@ func (device *Device) IpcGetOperation(w io.Writer) error {
 
 		for _, peer := range device.peers.keyMap {
 			// Serialize peer state.
-			// Do the work in an anonymous function so that we can use defer.
-			func() {
```

Comment:
i don't love the anonymous function a lot, but the defer prevents us from deadlocking on a panic from the send path. we shouldn't be panic'ing down there anymore, but I have a vague memory of popping out in a panic somewhere around here during shutdown, and needing to ensure that we unlock - out of an abundance of caution i'd probably retain the defer semantics

Reply:
Updated https://github.com/tailscale/wireguard-go/compare/d92db3d642d35e93c29f3d65424ecce585cb8e5e..97f4454d0e503d611d76bbdeca357efe5ab4dfaf

Reply:
oh, i think what i was remembering may have predated the buffer that sendf stages into, this might be more work than worthwhile now